server/src/main/java/io/deephaven/server/console/ConsoleServiceGrpcImpl.java

```diff
@@ -200,6 +180,26 @@ public void executeCommand(ExecuteCommandRequest request, StreamObserver<Execute
         });
     }
 
+    @Override
+    public void getHeapInfo(GetHeapInfoRequest request, StreamObserver<GetHeapInfoResponse> responseObserver) {
+        GrpcUtil.rpcWrapper(log, responseObserver, () -> {
+            final SessionState session = sessionService.getCurrentSession();
+            session.nonExport()
+                    .onError(responseObserver)
+                    .submit(() -> {
+                        GetHeapInfoResponse.Builder infoResponse = GetHeapInfoResponse.newBuilder();
+                        HeapInfo.Builder heapInfo = HeapInfo.newBuilder();
+                        heapInfo.setTotalHeapSize(RuntimeMemory.getInstance().totalMemory());
```

Comment:
just call getInstance() once?

And maybe chain the builder calls, but that's up to you